NextcloudTalkUITests/NextcloudTalkUITests.swift

```diff
@@ -0,0 +1,161 @@
+//
+// Copyright (c) 2023 Marcel Müller <marcel-mueller@gmx.de>
+//
+// Author Marcel Müller <marcel-mueller@gmx.de>
+//
+// GNU GPL version 3 or any later version
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+import XCTest
+
+final class NextcloudTalkUITests: XCTestCase {
+
+    let timeoutLong: Double = 60
+    let timeoutShort: Double = 15
+    let server = "http://localhost:8080"
+    let username = "admin"
+    let password = "admin"
+
+    override func setUpWithError() throws {
+        // In UI tests it is usually best to stop immediately when a failure occurs.
+        continueAfterFailure = false
+    }
+
+    override func tearDownWithError() throws {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+    }
+
+    func waitForEnabled(object: Any?) {
+        let enabledPredicate = NSPredicate(format: "enabled == true")
+        expectation(for: enabledPredicate, evaluatedWith: object, handler: nil)
+        waitForExpectations(timeout: timeoutLong, handler: nil)
+    }
+
+    func waitForHittable(object: Any?) {
+        let enabledPredicate = NSPredicate(format: "hittable == true")
+        expectation(for: enabledPredicate, evaluatedWith: object, handler: nil)
+        waitForExpectations(timeout: timeoutLong, handler: nil)
+    }
+
+    func launchAndLogin() -> XCUIApplication {
```

Comment:
I think a more descriptive format for test functions is needed. For example:
```
test_functionThatIsTested_withCircumstances_shouldExpectation()
```

so in this case it can become something like:

```
test_logIn_withProperParams_shouldLogInSuccessfully()
```